From 80fe263b49f7c3ea0d346cfbe9319bf08d47fe10 Mon Sep 17 00:00:00 2001
From: Juan Gutierrez <juan.gutierrez@nxp.com>
Date: Mon, 26 Sep 2016 16:51:06 -0500
Subject: [PATCH 1248/5242] MLK-13280 ARM: imx: missing mmdc read fifo reset
 on resume

commit  e347789054c2c50e765a1106a89eea5a58d98548 from
https://source.codeaurora.org/external/imx/linux-imx.git

When a device like (USB, CAMM, tty, etc) prevents the megamix
domain to power down during the suspend process (by enabling
a wakeup source) the resume process goes through a path where
the MMDC context should not be restored. However this resume
path does not reset the read fifo MMDCx_MPDGCTRL0[RST_RD_FIFO]
for the platforms with LPDDR2 causing a bad resuming and reset
of the device due to an exception.

This patch adds the reset_read_fifo on the No-restoring-MMDC
path to fix the bad resuming.

Signed-off-by: Juan Gutierrez <juan.gutierrez@nxp.com>
Signed-off-by: Alejandro Sierra <b18039@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/mach-imx/suspend-imx6.S |    1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/mach-imx/suspend-imx6.S b/arch/arm/mach-imx/suspend-imx6.S
index c50fefc..b8ad48f 100644
--- a/arch/arm/mach-imx/suspend-imx6.S
+++ b/arch/arm/mach-imx/suspend-imx6.S
@@ -300,6 +300,7 @@
 	bne	13f
 12:
 	resume_iomuxc_gpr
+	reset_read_fifo
 
 	b	17f
 13:
-- 
1.7.9.5

